temporalcli/strings.go

```diff
@@ -33,6 +33,40 @@ func (s *StringEnum) Set(p string) error {
 
 func (*StringEnum) Type() string { return "string" }
 
+type StringEnumArray struct {
+	// maps lower case value to original case
+	Allowed map[string]string
+	// values in original case
+	Values []string
+}
+
+func NewStringEnumArray(allowed []string, values []string) StringEnumArray {
+	// store allowed values in lower case so we can do case-insensitive comparison
+	var allowedMap = make(map[string]string)
+	for _, str := range allowed {
+		allowedMap[strings.ToLower(str)] = str
+	}
+
+	return StringEnumArray{Allowed: allowedMap, Values: values}
+}
+
+func (s *StringEnumArray) String() string { return strings.Join(s.Values, ",") }
+
+func (s *StringEnumArray) Set(p string) error {
+	val, ok := s.Allowed[strings.ToLower(p)]
+	if !ok {
+		values := make([]string, 0, len(s.Allowed))
+		for _, v := range s.Allowed {
+			values = append(values, v)
+		}
```

Comment:
just a suggestion if you like
```suggestion
		values := maps.Values(s.Allowed)
```
(from `"golang.org/x/exp/maps"`)

Reply:
> from "golang.org/x/exp/maps"

I think it's in the standard library now w/ the iter stuff

Reply:
They didn't include Keys and Values from x/exp/maps in the stdlib package. I'm not sure why, they're pretty useful :shrug: